src/transforms/transform-deployment.ts

```diff
@@ -120,6 +142,17 @@ export const mapEnvironment = (environment: string): string => {
 		return envNamesPattern.test(deburr(environment));
 	};
 
+	// if there is a user-defined config, we use that config for the mapping
+	if (config) {
+		const environmentType = mapEnvironmentWithConfig(environment, config);
+		if (environmentType){
+			return environmentType;
+		}
+	}
+
```

Comment:
shouldn't the user config be merge with the default?  

By doing it this way, it means that simply having a config yaml file with a single environment in `environmentMapping` will remove all defaults mappings for other enviroment.  ie. if I add `staging` as an environment only in the yaml, nothing will ever be mapped to production, dev, or testing.  This seems counter intuitive to me and will result in data loss for some customers

Reply:
Good call, and I want to clarify as well, when the repo has multi branch, which branch "wins" in terms of this config?

Reply:
@mboudreau The merge you had previously built failed because the hardcoded fallback is matching regexes whereas the user config matches glob patterns (because we don't want to have user-defined regexes that could be evil). We can't treat them the same, so I split it up.

But the case you mention is still covered by this logic. If the manual config doesn't provide a match, we look in the hardcoded config. I added a test case to showcase this (see test `falls back to hardcoded config when user config doesn't match`).